src/control/settings/ButtonConfig.cpp

```diff
@@ -70,17 +70,20 @@ bool ButtonConfig::applyNoChangeSettings(ToolHandler* toolHandler, Button button
         return false;
     }
     if (this->action == TOOL_PEN || this->action == TOOL_HIGHLIGHTER || this->action == TOOL_ERASER) {
-
-        if (this->drawingType == DRAWING_TYPE_DONT_CHANGE) {
-            toolHandler->setButtonDrawingType(toolHandler->getDrawingType(SelectedTool::toolbar), button);
-        }
+        Tool& correspondingTool = toolHandler->getTool(this->action);
```

Comment:
@rolandlo : This is the key as the settings should be taken from the corresponding Tool not just the currently active tool